lib/src/linting/rules/no-nullable-fields-within-requests.ts

```diff
@@ -0,0 +1,27 @@
+import { flatten, flow } from "lodash";
+import { LintingRule } from "../rule";
+
+import { TypeNode } from "../../models/nodes";
+import { isNullType, UnionType } from "../../models/types";
+import { extractRequestType } from "../../utilities/extract-endpoint-types";
+import { extractNestedUnionTypes } from "../../utilities/extract-nested-types";
+
+/**
+ * Checks optional fields at any level in requests are not nullable
+ */
+export const noNullableFieldsWithinRequests: LintingRule = contract => {
+  const extractTypes = flow(
+    extractRequestType,
+    (t: TypeNode) => extractNestedUnionTypes(t, contract.types),
+    flatten
+  );
+
+  const types = flatten(contract.endpoints.map(extractTypes));
+
+  const filterNullable = (t: TypeNode<UnionType>) =>
```

Comment:
Technically this doesn't do any filtering, it just returns a truthy value or not. The filtering occurs on the next line. Maybe call it `nullablePredicate`?